components/ai_chat/core/browser/conversation_driver_unittest.cc

```diff
@@ -983,4 +988,133 @@ TEST_P(PageContentRefineTest, TextEmbedderInitialized) {
+      .WillByDefault(
+          testing::Return(GURL("https://search.brave.com/search?q=test")));
+  EXPECT_CALL(*conversation_driver_, FetchSearchQuerySummary)
+      .WillOnce(base::test::RunOnceCallback<0>(
+          SearchQuerySummary("query", "summary")));
+  MockConversationDriverObserver observer(conversation_driver_.get());
+  EXPECT_CALL(observer, OnHistoryUpdate()).Times(1);  // From fetch.
+
+  // User opting in would trigger MaybeFetchSearchQuerySummary being
+  // called.
+  EmulateUserOptedIn();
+  EXPECT_TRUE(conversation_driver_->is_conversation_active_);
+  EXPECT_TRUE(conversation_driver_->should_send_page_contents_);
+
+  task_environment_.RunUntilIdle();
+  testing::Mock::VerifyAndClearExpectations(conversation_driver_.get());
+  testing::Mock::VerifyAndClearExpectations(&observer);
+
+  auto& history = conversation_driver_->GetConversationHistory();
+  std::vector<mojom::ConversationTurnPtr> expected_history;
+  expected_history.push_back(mojom::ConversationTurn::New(
+      mojom::CharacterType::HUMAN, mojom::ActionType::QUERY,
+      mojom::ConversationTurnVisibility::VISIBLE, "query", std::nullopt,
+      std::nullopt, base::Time::Now(), std::nullopt));
+  std::vector<mojom::ConversationEntryEventPtr> events;
+  events.push_back(mojom::ConversationEntryEvent::NewCompletionEvent(
+      mojom::CompletionEvent::New("summary", true)));
+  expected_history.push_back(mojom::ConversationTurn::New(
+      mojom::CharacterType::ASSISTANT, mojom::ActionType::RESPONSE,
+      mojom::ConversationTurnVisibility::VISIBLE, "summary", std::nullopt,
+      std::move(events), base::Time::Now(), std::nullopt));
+  ASSERT_EQ(history.size(), expected_history.size());
+  for (size_t i = 0; i < history.size(); i++) {
+    expected_history[i]->created_time = history[i]->created_time;
+    EXPECT_EQ(history[i], expected_history[i]);
+  }
+
+  // 2. Fetch again should have previous staged query and summary cleared.
+  EXPECT_CALL(*conversation_driver_, FetchSearchQuerySummary)
+      .WillOnce(base::test::RunOnceCallback<0>(
+          SearchQuerySummary("query", "summary")));
+  EXPECT_CALL(observer, OnHistoryUpdate()).Times(2);  // From clear and fetch.
+  conversation_driver_->MaybeFetchSearchQuerySummary();
+  task_environment_.RunUntilIdle();
+  testing::Mock::VerifyAndClearExpectations(conversation_driver_.get());
+  testing::Mock::VerifyAndClearExpectations(&observer);
+  ASSERT_EQ(history.size(), expected_history.size());
+  for (size_t i = 0; i < history.size(); i++) {
+    expected_history[i]->created_time = history[i]->created_time;
+    EXPECT_EQ(history[i], expected_history[i]);
+  }
+
+  // Helper function to push the query summary back to the conversation for
+  // further tests.
+  auto push_query_summary_to_history = [&]() {
+    for (auto& turn : expected_history) {
+      conversation_driver_->chat_history_.push_back(turn.Clone());
+    }
+  };
+
+  // 3. Fetch returns null result.
```

Comment:
Shouldn't these be separate tests instead of putting everything in one test and resetting state manually in-between each group - doesn't it run the risk of test contamination?

Reply:
Addressed in https://github.com/brave/brave-core/pull/25257/commits/3b1eaba8c93a65f967b833ab7168810164e63424